web/packages/teleterm/src/ui/StatusBar/StatusBar.tsx

```diff
@@ -17,15 +17,27 @@
  */
 
 import { Fragment } from 'react';
+import { useTheme } from 'styled-components';
 
-import { Flex, Text } from 'design';
+import { ButtonPrimary, Flex, Text } from 'design';
+
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import { ConnectionStatusIndicator } from 'teleterm/ui/TopBar/Connections/ConnectionsFilterableList/ConnectionStatusIndicator';
 
 import { AccessRequestCheckoutButton } from './AccessRequestCheckoutButton';
 import { ShareFeedback } from './ShareFeedback';
 import { useActiveDocumentClusterBreadcrumbs } from './useActiveDocumentClusterBreadcrumbs';
 
-export function StatusBar() {
+export function StatusBar(props: { onAssumedRolesClick(): void }) {
   const breadcrumbs = useActiveDocumentClusterBreadcrumbs();
+  const ctx = useAppContext();
+  const theme = useTheme();
+  const rootClusterUri = ctx.workspacesService.getRootClusterUri();
+
+  const assumedRoles = Object.values(
+    ctx.clustersService.getAssumedRequests(rootClusterUri)
```

Comment:
`StatusBar` gets updated when an access request gets assumed only because `useActiveDocumentClusterBreadcrumbs` subscribes to changes on `clustersService.state.clusters.get(clusterUri)`.

Should we use `useStoreSelector` here instead of depending on a `clustersService` method?

In the future, I feel like over time we'll have to migrate from using those methods to standalone functions that operate on state. In the meantime, we could make those methods call those functions instead (e.g. `ClustersService.prototype.getAssumedRequests` calls `getAssumedRequests` underneath which takes the state object).